### PR TITLE
gh-111374: Add a new PYTHON_FROZEN_MODULES env var, equivalent of `-X frozen_modules`.

### PR DESCRIPTION
**Pull request by yilei**
It seems that `config_get_env` can't distinguish between "not set" and "set to an empty string"? So `PYTHON_FROZEN_MODULES=` means differently than `-X frozen_modules` without values. I'm not sure if this could create confusion, but on the other hand the use of `-X frozen_modules` or `-X frozen_modules=` with empty values isn't explicitly documented.

* Issue: gh-111374

----
:books: Documentation preview :books:: https://cpython-previews--111411.org.readthedocs.build/

### CODE REVIEW

File: Doc/using/cmdline.rst [hunk 1]
```diff
@@ -1091,6 +1092,19 @@ conflict.
 
    .. versionadded:: 3.13
 
+.. envvar:: PYTHONFROZENMODULES
```
Comment (gpshead) [inline Doc/using/cmdline.rst]: as Hugo noted, we recently decided that new env vars should have `_` word separators, lets do that for this one.

Comment (yilei) [reply Doc/using/cmdline.rst]: Updated to `PYTHON_FROZEN_MODULES`, please take a look.

**Comment by hugovk** [comment]
Please also add this to https://docs.python.org/3.13/whatsnew/3.13.html

File: Doc/whatsnew/3.13.rst [hunk 2]
```diff
@@ -120,6 +120,11 @@ Other Language Changes
   is rejected when the global is used in the :keyword:`else` block.
   (Contributed by Irit Katriel in :gh:`111123`.)
 
+* Added a new environment variable :envvar:`PYTHON_FROZEN_MODULES`. It
+  determines whether or not frozen modules are ignored by the import machinery,
+  equivalent of the :option:`-X frozen_modules <-X>` command line option.
```
Comment (hugovk) [inline Doc/whatsnew/3.13.rst]:
```suggestion
  equivalent of the :option:`-X frozen_modules <-X>` command-line option.
```

File: Misc/NEWS.d/next/Core and Builtins/2023-10-27-11-22-09.gh-issue-111374.e9lrPZ.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Added a new environment variable :envvar:`PYTHON_FROZEN_MODULES`. It
+determines whether or not frozen modules are ignored by the import machinery,
+equivalent of the :option:`-X frozen_modules <-X>` command line option.
```
Comment (hugovk) [inline Misc/NEWS.d/next/Core and Builtins/2023-10-27-11-22-09.gh-issue-111374.e9lrPZ.rst]:
```suggestion
equivalent of the :option:`-X frozen_modules <-X>` command-line option.
```

File: Doc/using/cmdline.rst [hunk 4]
```diff
@@ -1091,6 +1092,19 @@ conflict.
...
    .. versionadded:: 3.13
 
+.. envvar:: PYTHON_FROZEN_MODULES
+
+   If this variable is set to ``on`` or ``off``, it determines whether or not
+   frozen modules are ignored by the import machinery.  A value of "on" means
+   they get imported and "off" means they are ignored.  The default is "on"
+   for non-debug builds (the normal case) and "off" for debug builds.
+   Note that the "importlib_bootstrap" and "importlib_bootstrap_external"
+   frozen modules are always used, even if this flag is set to "off".
```
Comment (hugovk) [inline Doc/using/cmdline.rst]:
```suggestion
   If this variable is set to ``on`` or ``off``, it determines whether or not
   frozen modules are ignored by the import machinery.  A value of ``on`` means
   they get imported and ``off`` means they are ignored.  The default is ``on``
   for non-debug builds (the normal case) and ``off`` for debug builds.
   Note that the :mod:`!importlib_bootstrap` and :mod:`!importlib_bootstrap_external`
   frozen modules are always used, even if this flag is set to ``off``.
```

And we can apply the same formatting to the ``-X frozen_modules`` section above.

**Review by hugovk — COMMENTED** [review]
Thanks for the docs changes. Looks good, just some little nits.

**Comment by yilei** [comment]
Thanks for the review, `-X frozen_modules` is also updated. Please take another look.

**Review by gpshead — APPROVED** [review]

**Pull request merged**